### DISCUSSION TITLE
Is there any documentation on using nuclei as go library?

### DISCUSSION THREAD

**Discussion by mikerott** [projectdiscovery/nuclei#1520] [Q&A] [answered]
Hi, I have a project where I'd like to wrap nuclei behind an http service implementing a job queue.  To do so, I want to use nuclei as a library.  I've seen prior discussion ([here](https://github.com/projectdiscovery/nuclei/discussions/546)) and the mention of using it as a lib ([here](https://github.com/projectdiscovery/nuclei/issues/1152) which links milestone 2.5.4 which I assume correlates to [release 2.5.4](https://github.com/projectdiscovery/nuclei/releases/tag/v2.5.4)).

That said, I see no usage examples or documentation about how to actually use it as a library.  I'd rather not plow through [main.go](https://github.com/projectdiscovery/nuclei/blob/master/v2/cmd/nuclei/main.go) to reverse engineer it.  Can you help with a pointer to some documentation?  Or even a unit test I should study?  Maybe a quick typed up example?  What would be my main entry point for using nuclei as an embedded go library?

**Comment by ehsandeep** [comment, marked as answer]
@mikerott please refer to https://github.com/projectdiscovery/nuclei/blob/master/DESIGN.md#using-nuclei-from-go-code

**Reply by mikerott** [reply]
Thank you so much!  I guess I hadn't visited enough of the docs or somehow my eyes missed it.
